### ISSUE TITLE
DDLDatabase can not handle multiple constraints with the same name

### ISSUE THREAD

**Issue by vojtapol** [jOOQ/jOOQ#8824]
This MySQL script executes without errors on MySQL:

```SQL
CREATE TABLE `BATCH_JOB_EXECUTION_SEQ` (
  `ID`         BIGINT NOT NULL,
  `UNIQUE_KEY` CHAR   NOT NULL,
  CONSTRAINT `UNIQUE_KEY`
    UNIQUE (`UNIQUE_KEY`)
);

CREATE TABLE `BATCH_JOB_SEQ` (
  `ID`         BIGINT NOT NULL,
  `UNIQUE_KEY` CHAR   NOT NULL,
  CONSTRAINT `UNIQUE_KEY`
    UNIQUE (`UNIQUE_KEY`)
);
```

When used as a script for `jooq-codegen-maven` with `DDLDatabase` it throws this error:

```
Error running jOOQ code generation tool: Error while exporting schema: 

SQL [create table "BATCH_JOB_SEQ"("ID" bigint not null, "UNIQUE_KEY" char not null, constraint "UNIQUE_KEY" unique ("UNIQUE_KEY"))]; Constraint "UNIQUE_KEY" already exists; 

SQL statement: create table "BATCH_JOB_SEQ"("ID" bigint not null, "UNIQUE_KEY" char not null, constraint "UNIQUE_KEY" unique ("UNIQUE_KEY")) [90045-197]
```

The error `90045-197` refers to an H2 error that is thrown when you try to create a second constraint with the same name. This is not a problem for MySQL though so jOOQ should handle this case.

I am not sure if jOOQ uses the constraint names for anything but I would be fine if they were replaced by something random so that H2 can handle it.

**Comment by lukaseder** [comment]
Thanks a lot for your report. I see, this is indeed unfortunate. Could happen with SQL Server as well, which also has per-table constraint namespacing.

Of course, I personally think it is generally a good idea to have unique constraint names within a schema. I always use the table name somewhere in the constraint name, e.g. `batch_job_seq_uk`, possibly suffixed with column names. But this doesn't help you right now.

A workaround is to preprocess your SQL script prior to passing it to the `DDLDatabase`, and remove all ``CONSTRAINT `UNIQUE_KEY` `` declarations (or comment them out), effectively making your constraints unnamed. For example:

```sql
CREATE TABLE `BATCH_JOB_SEQ` (
  `ID`         BIGINT NOT NULL,
  `UNIQUE_KEY` CHAR   NOT NULL,
  -- CONSTRAINT `UNIQUE_KEY`
    UNIQUE (`UNIQUE_KEY`)
);
```

Also, in jOOQ 3.12, we've introduced a new parser specific comment syntax (see https://github.com/jOOQ/jOOQ/issues/8325). Using this syntax, you can add comments that are understood only by jOOQ's parser, and ignored by all other SQL clients, such as:

```sql
CREATE TABLE `BATCH_JOB_SEQ` (
  `ID`         BIGINT NOT NULL,
  `UNIQUE_KEY` CHAR   NOT NULL,
  -- [jooq ignore start]
  CONSTRAINT `UNIQUE_KEY`
  -- [jooq ignore end]
    UNIQUE (`UNIQUE_KEY`)
);
```

The jOOQ parser will then effectively not "see" anything between the two `[jooq ignore start]` and `[jooq ignore end]` comment tokens.

I will, in the meantime, think about how to work around this problem from within jOOQ. The best option here, would be, of course, not to rely on H2: https://github.com/jOOQ/jOOQ/issues/7034

**Comment by lukaseder** [comment]
See also: #6232

**Comment by lukaseder** [comment]
This works correctly with our new DDL interpreter based implementation of the `DDLDatabase`. I'm not sure if we have to fix this for the current H2 based implementation, which is limited by the feature set of H2.

**Comment by vojtapol** [comment]
In my opinion it does not have to be supported in H2. In which version is the new DDL interpreter going to be released?

**Comment by lukaseder** [comment]
We won't fix this in the H2 implementation for 3.13. Our own DDL interpreter can handle it.
